### DISCUSSION TITLE
Refresh page on new release if used components were changed

### DISCUSSION THREAD

**Discussion by lorddaedra** [django-commons/django-unicorn#241] [General]
I'm not yet sure but looks like every new release of project with django-unicorn components require to refresh page if components were changed. 

Is it something broken by design for all server-side component systems or we can fix it?..

Can we somehow tell user about it (right top or bottom banner): "Hey! We have updated panel, so you use outdated frontend, please, refresh page now to update it. <refresh button>"

I think it's needed for any serious production usage of django-unicorn.

**Comment by adamghill** [comment]
I would think this same issue would occur for any SPA (or long-lived application that stores state on the client, right? I just searched around a little bit and found this for Vue: https://medium.com/@codemonk/automatic-reload-of-client-after-deploying-in-vue-js-91c120f85f0e. Off the top of my head, the current deployment could be calculated somehow and then a `poll` could check for a change in the deployment SHA?

**Reply by lorddaedra** [reply]
Sorry for the delay, it was a hot week..

I thinks it's a problem of most reactive frameworks by default. After server side component updates we need somehow update client side too (or at least send message for user to do it manually). 

In classic SPA usually we have frontend client part (js framework with VueJS/React components in browser) and backend server part (Django + DRF) and API contract between them (OpenAPI based, for example). 

We can maintain several API versions manually to make all server things compatible with old set of frontend components and with new set components. But in case of reactive frameworks like django-unicorn we need some help to do it because of in case of reactive frameworks it a little bit harder to monitor contract changes, we have server-side frontend and client-side frontend and know almost nothing about their communication (some low-level magic). 

AFAIK this is one of the reasons why some companies do not like to use reactive frameworks. Assume, you have several Kubernetes containers and start update process. How not to break clients if part of containers use 1.0 version and part of them use 2.0? May be it's issue of all reactive frameworks by design...

I think automatic [in browser] client side reload may help to reduce damage. It's better than nothing. So yes, we can check experience of SPA frameworks here...
